### PR TITLE
cat: Unrevert splice patch

### PR DESCRIPTION
**Pull request by ArniDagur**

### CODE REVIEW

**Comment by sylvestre** [comment]
Code coverage isn't:
```

   Compiling uu_stdbuf v0.0.6 (/home/runner/work/coreutils/coreutils/src/uu/stdbuf)
error[E0277]: can't compare `[u8]` with `std::vec::Vec<u8>`
Error:    --> /home/runner/work/coreutils/coreutils/tests/by-util/test_cat.rs:321:5
    |
321 |     assert_eq!(&buf[..], &expected);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no implementation for `[u8] == std::vec::Vec<u8>`
    |
    = help: the trait `std::cmp::PartialEq<std::vec::Vec<u8>>` is not implemented for `[u8]`
    = note: required because of the requirements on the impl of `std::cmp::PartialEq<&std::vec::Vec<u8>>` for `&[u8]`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by ArniDagur** [comment]
The test errors are fixed now. They were due to my use of a compiler newer than 1.40.0 for development.

File: tests/by-util/test_cat.rs [hunk 1]
```diff
@@ -301,7 +301,7 @@ fn test_dev_full() {
     let mut proc_stdout = proc.stdout.take().unwrap();
     let expected = [0; 2048];
     proc_stdout.read_exact(&mut buf).unwrap();
-    assert_eq!(&buf[..], &expected);
+    assert_eq!(&buf[..], &expected[..]);
```
Comment (sylvestre) [inline tests/by-util/test_cat.rs:304]: please run rustfmt

Comment (ArniDagur) [reply tests/by-util/test_cat.rs:304]: Done.

File: Cargo.lock [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
+version = 3
```
Comment (sylvestre) [inline Cargo.lock:3]: can you revert this?

File: Cargo.lock [hunk 3]
```diff
@@ -1,6 +1,5 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
-version = 3
```
Comment (sylvestre) [inline Cargo.lock]: the proper way to update it is:
cargo +1.40.0 update

**Comment by sylvestre** [comment]
Well done: 
much better than the gnu version
```
  'coreutils/target/release/coreutils cat a_large_file.mov   > /dev/null' ran
    1.42 ± 0.08 times faster than 'cat cat a_large_file.mov> /dev/null'
    1.56 ± 0.13 times faster than 'coreutils-old cat cat a_large_file.mov  > /dev/null'
```

**Pull request merged**